drivers/i2c/i2c_nrfx_twi_rtio.c

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (c) 2018, Nordic Semiconductor ASA
+ * Copyright (c) 2024, Croxel Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/i2c/rtio.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <soc.h>
+#include <nrfx_twi.h>
+
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+LOG_MODULE_REGISTER(i2c_nrfx_twi, CONFIG_I2C_LOG_LEVEL);
+
+#if CONFIG_I2C_NRFX_TRANSFER_TIMEOUT
+#define I2C_TRANSFER_TIMEOUT_MSEC K_MSEC(CONFIG_I2C_NRFX_TRANSFER_TIMEOUT)
+#else
+#define I2C_TRANSFER_TIMEOUT_MSEC K_FOREVER
+#endif
+
+struct i2c_nrfx_twi_data {
+	struct i2c_rtio *ctx;
+	struct k_sem transfer_sync;
+	struct k_sem completion_sync;
+	volatile nrfx_err_t res;
```

Comment:
Is this still needed or is the error returnable more directly with rtio_iodev_sqe_err()?

Reply:
No longer needed. Removed!